File: fixed_heap_dynamic.py
def fixed_heap_dynamic():
    # Fixed-size list created using list() — heap-allocated
    arr = list(range(5))  # [0, 1, 2, 3, 4]

    # Modify contents
    for i in range(len(arr)):
        arr[i] = (i + 1) * 10

    # Display the list
    print("Fixed Heap-Dynamic Array:")
    print(arr)


fixed_heap_dynamic()

File: fixed_heap_dynamic.java
public class FixedHeapDynamicExample {

    public static void main(String[] args) {
        // Fixed-size array allocated on the heap
        Integer[] fixedHeapArray = new Integer[4]; // allocated with 'new'

        // Assign values
        for (int i = 0; i < fixedHeapArray.length; i++) {
            fixedHeapArray[i] = (i + 1) * 100;
        }

        // Display the array
        System.out.println("Fixed Heap-Dynamic Array:");
        for (Integer value : fixedHeapArray) {
            System.out.print(value + " ");
        }
    }
}
